nrf_security/Kconfig.psa

```diff
@@ -78,6 +78,14 @@ config MBEDTLS_USE_PSA_CRYPTO
 	help
 	  Corresponds to MBEDTLS_USE_PSA_CRYPTO setting in mbed TLS config file.
 
+config PSA_EITS_BACKEND_TFM
```

Comment:
The naming "BACKEND" is unfortunate as the term is used for PSA_NATIVE_ITS and this configurations is not grouped within. 

I propose maybe PSA_EITS or PSA_ITS_ENCRYPTED to normalize this. If we state PSA_EITS/PSA_ITS_ENCRYPTED we can potentially reuse this for non TF-M usage as well.

Reply:
Adding something as generic as PSA_ITS_ENCRYPTED/PSA_EITS  when we have the NATIVE_ITS I think that is more confusing to the users. Adding a generic PSA_ITS_ENCRYPTED/PSA_EITS as a top level for every use case and independent use cases for backends makes more sense to me. What do you think?

Reply:
The TFM config for this is currently TFM_ITS_ENCRYPT, which I'm proposing is changed to TFM_ITS_ENCRYPTED. This would in my opinion not be difficult to understand. It is also possible to make dependent defaults:

`config XXXX_ITS_ENCRYPTED
    default Y if YYYY_IS_ENCRYPTED`

To keep the internal consistency between PSA and TFM configurations. 

The term backend is a legacy term that we are gravitating away from because of the addition of the term "driver" in PSA APIs.

Reply:
That's a valid point, this looks better in the Kconfigs for sure. I will update this.